	* src/cairo-wideint-private.h: Add missing definitions on Tru64

	* test/pdiff/pdiff.c: Fix compilation on Solaris, Tru64 and HP-UX

--- test/pdiff/pdiff.c.old	2007-03-23 15:10:23.226278183 +0100
+++ test/pdiff/pdiff.c	2007-03-23 15:11:18.486787797 +0100
@@ -20,7 +20,13 @@
 #include <math.h>
 #include <stdio.h>
 #include <stdlib.h>
+
+#ifdef sun
+#include <sys/int_types.h>
+#elif !defined (__hpux__) && !defined (__osf__)
 #include <stdint.h>
+#endif
+
 #include "pdiff.h"
 
 #ifndef M_PI
--- src/cairo-wideint-private.h.old	2007-03-26 16:16:15.012047000 -0400
+++ src/cairo-wideint-private.h	2007-03-26 16:22:16.000333000 -0400
@@ -68,6 +68,22 @@
 #error Cannot find definitions for fixed-width integral types (uint8_t, uint32_t, etc.)
 #endif
 
+# ifndef INT16_MIN
+#  define INT16_MIN     (-32767-1)
+# endif
+# ifndef INT16_MAX
+#  define INT16_MAX     (32767)
+# endif
+# ifndef UINT16_MAX
+#  define UINT16_MAX    (65535)
+# endif
+# ifndef INT32_MIN
+#  define INT32_MIN     (-2147483647-1)
+# endif
+# ifndef INT32_MAX
+#  define INT32_MAX     (2147483647)
+# endif
+
 /*
  * 64-bit datatypes.  Two separate implementations, one using
  * built-in 64-bit signed/unsigned types another implemented
